context-js/src/lib.rs

```diff
@@ -175,3 +175,28 @@ pub async fn parse_internal_bin_and_meta_from_tarball(
         versioned_bundle,
     })
 }
+
+#[wasm_bindgen]
+// trunk-ignore(clippy/too_many_arguments)
+pub fn gen_info_id(
+    org_url_slug: String,
+    repo_full_name: String,
+    file: Option<String>,
+    classname: Option<String>,
+    parent_fact_path: Option<String>,
+    name: Option<String>,
```

Comment:
I switch from `&str` to `String` since I'm getting errors that wasm doesn't know how to create an equivalent for `&str` but it did for `String` - if there was something else I should have done please let me know